eng/scripts/npm-admin-tasks.ps1

```diff
@@ -0,0 +1,64 @@
+param (
+  [Parameter(mandatory = $true)]
+  $taskType,
+  [Parameter(mandatory = $true)]
+  $packageName,
+  [Parameter(mandatory = $true)]
+  $pkgVersion,
+  $tagName,
+  [Parameter(mandatory = $true)]
+  $npmToken,
+  $reason
+)
+
+try {
+  Write-Host "Setting AuthToken Deployment"
+  $env:NPM_TOKEN = $npmToken
+  $regAuth = "//registry.npmjs.org/"
+  npm config set $regAuth`:_authToken=`$`{NPM_TOKEN`}
+  $nameAndVersion = $packageName + "@" + $pkgVersion
+
+  # Verify that package name is not "@azure"
+  if ($packageName -eq '@azure') {
+    Write-Host "Invalid package name"
+    exit 1
+  }
+
+  switch ($taskType) {
+    "AddTag" {
+      Write-Host "Adding tag for package"
+      Write-Host "npm dist-tag add $($nameAndVersion) $tagName"
+      npm dist-tag add $nameAndVersion $tagName
+    }
+
+    "RemoveTag" {
+      Write-Host "Removing tag for package"
+      Write-Host "npm dist-tag rm $($nameAndVersion) $tagName"
+      npm dist-tag rm $nameAndVersion $tagName
+    }
+
+    "DeprecatePackage" {
+      if ($reason -eq '') {
```

Comment:
To un-deprecate a package, an empty string is used for the `message` argument to `npm deprecate`.  So maybe we should support an empty message for un-deprecate?  Or add another task type for un-deprecate?

Reply:
Thanks @mikeharder I will add a different task type `UnDeprecate` to keep it more clear.

Reply:
I haven't got un-deprecate part working yet. It's throwing error even with syntax as per npm documentation. I will submit a separate PR to add this option once I figure out the issue. I'll file an issue to track this.